CONTRIBUTING.md

```diff
@@ -0,0 +1,341 @@
+* unit testing of individual functions and methodsm and,
+* integration testing of using combinations of modules.
+
+These are the main tests that the package is behaving as expected and producing stable
+outputs and can be run from repository using:
+
+```bash
+poetry run pytest
+```
+
+#### Using `doctest`
+
+The package docstrings contain `doctest` examples of code use. These examples are
+intended to provide simple examples of method or function use and generate an output and
+the `doctest` process is used to validate that the provided values are correct.
+
+These tests are automatically run when `pytest` is run, but individual module files can
+be checked using, for example:
+
+```bash
+poetry run python -m doctest pyrealm/pmodel/pmodel.py
+```
+
+Normally, `doctest` is used to test a return value but can also be used to check that an
+error or warning is raised. See the example for `pyrealm.utilities.convert_rh_to_vpd`
+which redirects the stderr to stdout to allow for the warning text to be included in the
+doctest.
+
+### Building the documentation
+
+`sphinx` is used to build an HTML version of the package documentation provided in
+`docs/source` and to build the API documentation provided in the code docstrings. The
+`sphinx` building process requires some extra packages, but these are included in the
+`docs` group in `pyproject.toml` and should be installed.
+
+In order to build the package documentation, Jupyter needs to be able to associate the
+documentation files with the Python environment managed by `poetry`. This is done by
+installing the `poetry` environment as a new Jupyter kernel with a fixed name. This
+allows all build systems to run notebooks using the correct build environment:
+
+```bash
+# Set ipython kernel
+poetry run python -m ipykernel install --user --name=pyrealm_python3
+```
+
+In order to build the package documentation, the following command can be used:
+
+```bash
+# Build docs using sphinx
+cd docs
+poetry run sphinx-build -W --keep-going source build
+```
+
+### Online documentation
+
+The documentation for the package is hosted at:
+
+[https://pyrealm.readthedocs.io/en/develop/pmodel.html](https://pyrealm.readthedocs.io/en/develop/pmodel.html)
+
+This has been configured to automatically build commits to the `main` branch, which
+should generate version specific sets of documentation.
+
+### Referencing
+
+The documentation uses the `sphinxcontrib-bibtex` package to support citations. This
+uses Latex like citation keys in the documentation to insert references and build a
+bibliography. The `sphinx` configuration in `docs/source/conf.py` provides a custom
+Author/Year citation style. The reference library in `source/refs.bib` needs to be kept
+up to date with the literature for the project.
+
+## Release process
+
+Releasing a new version of the package follows the flow below:
+
+1. Create a `release` branch from `develop` containing the new release code.
```

Comment:
I like to use the github function to create a branch from an issue (on the right-hand-side under "development"), as this links the branch to the issue. We can discuss whether this is something you want to recommend in your codebase, or leave it up to the developer.

Reply:
Ah, ok, sorry, I misunderstood that.